jenkins/Hono-Website-Pipeline-Declarative.groovy

```diff
@@ -69,150 +69,137 @@ pipeline {
       cron('TZ=Europe/Berlin\n# every night between 3 and 4 AM\nH 3 * * *')
   }
 
+  environment {
+    HONO_HOMEPAGE_DIR="${WORKSPACE}/hono/site/homepage"
+    HONO_DOCUMENTATION_DIR="${WORKSPACE}/hono/site/documentation"
+    WEBSITE_TARGET_DIR="${WORKSPACE}/hono-web-site"
+    DOC_ASSEMBLY_DIR="${WORKSPACE}/hono-documentation-assembly"
+  }
+
   stages {
 
-    stage('Prepare workspace') {
+    stage("Prepare workspace") {
       steps {
-        echo "Cloning Hono repository..."
         sh '''
-          git clone https://github.com/eclipse-hono/hono.git ${WORKSPACE}/hono
-        '''
-        echo "Copying Documentation directory from master branch..."
-        sh ''' 
-           cp -r ${WORKSPACE}/hono/site/documentation ${WORKSPACE}/hono-documentation-assembly
-           mkdir -p ${WORKSPACE}/hono-documentation-assembly/content_dirs
+          #!/bin/bash
+          echo "cloning Hono repository..."
+          git clone --recurse-submodules https://github.com/eclipse-hono/hono.git "${WORKSPACE}/hono"
+          echo "copying Documentation directory from master branch..."
+          cp -r "${HONO_DOCUMENTATION_DIR}" "${DOC_ASSEMBLY_DIR}"
+          mkdir -p "${DOC_ASSEMBLY_DIR}/content_dirs"
         '''
       }
     }
 
-    stage('Cloning Hugo themes') {
+    stage("Clone Hono web site repository") {
       steps {
-        echo "cloning Hugo Relearn theme..."
-        sh '''
-            git clone --depth 1 --branch 4.2.5 https://github.com/McShelby/hugo-theme-relearn.git ${WORKSPACE}/hono-documentation-assembly/themes/hugo-theme-relearn
-        '''
-        echo "cloning Hugo Universal theme..."
-        sh '''
-            git clone --depth 1 --branch 1.3.2 https://github.com/devcows/hugo-universal-theme.git ${WORKSPACE}/hono/site/homepage/themes/hugo-universal-theme
-            echo "Remove images from theme" # We do not need the pictures. Removing them, so they don't get deployed
-            rm ${WORKSPACE}/hono/site/homepage/themes/hugo-universal-theme/static/img/*
-        '''
+        sshagent(["git.eclipse.org-bot-ssh"]) {
+          sh '''
+            #!/bin/bash
+            echo "cloning Hono web site repository..."
+            git clone ssh://genie.hono@git.eclipse.org:29418/www.eclipse.org/hono.git ${WEBSITE_TARGET_DIR}
+            echo "scrubbing web site target directory..."
+            rm -rf ${WEBSITE_TARGET_DIR}/*
+          '''
+        }
       }
     }
 
-    stage('Cloning Hono web site repository') {
+    stage("Build home page") {
       steps {
-        sshagent(["git.eclipse.org-bot-ssh"]) {
+        container("hugo") {
           sh '''
-            echo "cloning Hono web site repository..."
-            git clone ssh://genie.hono@git.eclipse.org:29418/www.eclipse.org/hono.git ${WORKSPACE}/hono-web-site
-            echo "scrubbing web site target directory..."
-            rm -rf ${WORKSPACE}/hono-web-site/*
+            #!/bin/bash
+            cd ${HONO_HOMEPAGE_DIR}
```

Comment:
Shellcheck findings: 
Use 'cd ... || exit' or 'cd ... || return' in case cd fails.
Double quote to prevent globbing and word splitting.

Also applies to the other lines below containing `cd`.

Reply:
The scripts will exit on any failure anyway because of the way the Jenkins pipeline plugin executes them ...